packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/EncryptionTests.kt

```diff
@@ -122,4 +131,59 @@ class EncryptionTests {
                 }
             }
     }
+
+    @OptIn(ExperimentalEncryptionCallbackApi::class)
+    @Test
+    fun openEncryptedRealmWithEncryptionKeyCallback() = runBlocking {
```

Comment:
It would also be nice with tests for invalid pointers. Do we throw a correct exception in that case or ends up with a native crash?

Reply:
Invalid pointers can cause a segmentation fault that will crash the entire test suite. I'll add a comment on the class doc about key validity.